src/main/groovy/com/entagen/jenkins/JenkinsApi.groovy

```diff
@@ -224,7 +224,11 @@ class JenkinsApi {
         http.handler.failure = { resp ->
             def msg = "Unexpected failure on $jenkinsServerUrl$path: ${resp.statusLine} ${resp.status}"
             status = resp.statusLine.statusCode
-            throw new Exception(msg)
+            if (status != 500) {
```

Comment:
The main issue we have is with jenkins locking build logs for the deploy job, which prevents the deletion of the folder which results in a 500 when we try to delete the job.